nuts/context.py

```diff
@@ -130,8 +130,18 @@ def general_result(self) -> AggregatedResult:
 
         if nornir_filter:
             selected_hosts = self.nornir.filter(nornir_filter)
+
+            if not selected_hosts.inventory.hosts:
+                raise NutsSetupError(
+                    f'Host(s) "{",".join(nornir_filter.filters["name__any"])}" '
+                    f"not found in the inventory."
+                )
         else:
             selected_hosts = self.nornir
+
+            if not selected_hosts.inventory.hosts:
+                raise NutsSetupError("No Hosts found, is the nornir inventory empty?")
```

Comment:
I think this check can be merged into one with a more generic error message, something like "No matching hosts found in the nornir inventory", otherwise it's just duplicated code and thus a code smell

Reply:
I'm a big fan of giving the user specific feedback, so I duplicated this by intention. I know what you mean, but I would like to leave this as I put it into the PR.

Reply:
In a direct discussion we agreed that we will split the filtering and the error handling, but keeping the split error handling. With this we can still provide the user with good feedback while not mixing the code concerns together.